spotify_player/src/cli/commands.rs

```diff
@@ -128,3 +128,43 @@ pub fn init_like_command() -> Command {
 pub fn init_authenticate_command() -> Command {
     Command::new("authenticate").about("Authenticate the application")
 }
+
+pub fn init_playlist_subcommand() -> Command {
+    Command::new("playlist")
+        .about("Playlist editing")
+        .subcommand_required(true)
+        .subcommand(Command::new("new").about("Create a new playlist")
+            .arg(Arg::new("name")
+                .value_parser(clap::builder::NonEmptyStringValueParser::new()))
+            .arg(Arg::new("description")
+                .value_parser(clap::builder::NonEmptyStringValueParser::new())
+                .required(false))
+            .arg(Arg::new("public")
+                .short('p')
+                .long("public")
+                .action(clap::ArgAction::SetTrue)
+                .help("Sets the playlist to public"))
+            .arg(Arg::new("collab")
+                .short('c')
+                .long("collab")
+                .action(clap::ArgAction::SetTrue)
+                .help("Sets the playlist to collaborative"))
+            )
+        .subcommand(Command::new("delete").about("Delete a playlist")
+            .arg(Arg::new("id")
+                .value_parser(clap::builder::NonEmptyStringValueParser::new())))
+        .subcommand(Command::new("import").about("Imports all songs from a playlist into another playlist.")
+            .arg(Arg::new("from")
+                .value_parser(clap::builder::NonEmptyStringValueParser::new()))
+            .arg(Arg::new("to")
+                .value_parser(clap::builder::NonEmptyStringValueParser::new()))
+            .after_help("Copies all tracks into a playlist. If imported again, it adds new tracks and outputs information on deleted tracks since last import."))
+        .subcommand(Command::new("list").about("Lists all user playlists."))
+        .subcommand(Command::new("fork").about("Creates a copy of a playlist and imports it.")
+            .arg(Arg::new("id")
+                .value_parser(clap::builder::NonEmptyStringValueParser::new())))
+        .subcommand(Command::new("update").about("Updates all imported playlists or a single individual import.")
```

Comment:
Haven't read the implementation but based on the command's description, I don't understand the purpose of this command.

Reply:
The update command is used to update imports. This just means that any imported playlists will be re imported. It is for ease of use. It would be more difficult for me to find out what was imported and do a command for each.

Reply:
Iiuc, for each playlist, you want to store a list of "from" playlists imported to that playlist. `update` command basically runs `import` with that playlist and each of the "from" playlists from the list.

tbh, I'm not too fond of the idea of having this as a CLI command as I don't find a lot of use cases for this feature.  You can create an external script to do that with the `import` command.

Reply:
Without the update command, managing imports would be tedious. It could definitely be a script but why should a user have to set up a script to keep their playlists imported?

I will and do use the update command the most. Imports/forks are done once but keeping them updated will be an everyday thing.